### DISCUSSION TITLE
Redux Crashes the app in React Native IOS Release Build

### DISCUSSION THREAD

**Discussion by tradebulls** [reduxjs/redux#4195] [Help]
We are facing crash in few of our IOS devices in production app having redux in our project. We are using following libraries in our project:

"react-redux": "7.2.5",
 "redux": "4.1.1",
"redux-thunk": "2.3.0"
"react-native":"0.63.4"

We tried replicating the same in latest version of react native as well i.e 0.66.1. The issue can be replicated simply by creating a new project and adding redux to it. It works fine in debug mode and crashes on launch in Release mode.

When we comment Provider tag in index.js the app works perfectly fine in debug as well as release build.

Also, this works fine on Android debug and release mode. Only issue is replicating in IOS release build only.

Any help? Thanks in advance

**Comment by tradebulls** [comment]
We tried Profile Building the app in Xcode, which helps us with the logs what crashes the app. The logs are as follows:

*** Terminating app due to uncaught exception 'RCTFatalException: Unhandled JS Exception: RangeError: Maximum call stack size exceeded.', reason: 'Unhandled JS Exception: RangeError: Maximum call stack size exceeded., stack:
v@2:1473
<unknown>@437:916
v@2:1473
<unknown>@436:117
v@2:1473
<unknown>@435:190
v@2:1473
<unknown>@6:69
v@2:1473
d@2:875
global code@1195:3
'
*** First throw call stack:
(0x1ab7b8300 0x1ab4ccc1c 0x1022fd6a4 0x102378918 0x102379240 0x1ab7bef14 0x1ab688098 0x1ab688cc0 0x10232ccf4 0x10232ee4c 0x10232ebbc 0x1ab456ec4 0x1ab45833c 0x1ab45e85c 0x1ab45f290 0x1ab468928 0x1ab4bf714 0x1ab4c59c8)


If we comment Provider tag in index.js then everything works fine. Please help

**Reply by markerikson** [reply]
I'm afraid that we have no way of knowing what's going on in your app, and no way of debugging something that's going on in React Native like that.  Sorry :(

**Comment by halaei** [comment]
@tradebulls same issue here. Did you find the cause or any fix for that?

**Reply by ShashwatMDas** [reply]
@tradebulls @halaei Facing the same issue here, were you able to find any solution?

**Reply by halaei** [reply]
Yes. Take a look at my comment: https://github.com/facebook/react-native/issues/32672#issuecomment-988851104
You probably need to split some javascript file with +500 exports in it, in our case a file in which we used to place all the action types.

**Comment by asingh-lifesize** [comment]
have anyone find a solution to this?

**Reply by phryneas** [reply]
Without a reproduction we couldn't even try to start looking.
